pyiceberg/table/metadata.py

```diff
@@ -517,12 +517,15 @@ def new_table_metadata(
     location: str,
     properties: Properties = EMPTY_DICT,
     table_uuid: Optional[uuid.UUID] = None,
+    assign_fresh_ids: bool = True,
 ) -> TableMetadata:
     from pyiceberg.table import TableProperties
 
-    fresh_schema = assign_fresh_schema_ids(schema)
-    fresh_partition_spec = assign_fresh_partition_spec_ids(partition_spec, schema, fresh_schema)
-    fresh_sort_order = assign_fresh_sort_order_ids(sort_order, schema, fresh_schema)
+    if assign_fresh_ids:
+        fresh_schema = assign_fresh_schema_ids(schema)
+        partition_spec = assign_fresh_partition_spec_ids(partition_spec, schema, fresh_schema)
+        sort_order = assign_fresh_sort_order_ids(sort_order, schema, fresh_schema)
+        schema = fresh_schema
```

Comment:
My proposal is to not include `assign_fresh_ids` as a flag in functions other than `new_table_metadata`. 
So when `_create_staged_table` and `create_table` is given
* a `pa.Schema`, convert to `Schema` and set `assign_fresh_ids` to True in `new_table_metadata`
* a `Schema`. Assume the user created `Schema` with the correct IDs (possibly verify some correctness characteristics such as uniqueness). And use the schema as is
If a user wants to reassign IDs for `Schema`, this can be done outside the `create_table` functions and we can  even provide a helper function to do so. 

I feel like this way can help break apart the responsibilities of schema id assignment from the `create_table` methods.

Reply:
LMK if this makes sense or if im missing something!